src/hotspot/share/memory/metaspace/metaspaceArena.cpp

```diff
@@ -365,7 +373,13 @@ void MetaspaceArena::deallocate_locked(MetaWord* p, size_t word_size) {
   UL2(trace, "deallocating " PTR_FORMAT ", word size: " SIZE_FORMAT ".",
       p2i(p), word_size);
 
+  // Only blocks that had been allocated via MetaspaceArena::allocate(size) must be handed in
+  // to MetaspaceArena::deallocate(), and only with the same size that had been original used for allocation.
+  // Therefore the pointer must be aligned correctly, and size can be alignment-adjusted (the latter
+  // only matters on 32-bit):
+  assert_is_aligned_metaspace_pointer(p);
   size_t raw_word_size = get_raw_word_size_for_requested_word_size(word_size);
+
```

Comment:
If this is deallocating a pointer returned from metaspace, why isn't raw_word_size the same as word_size?